angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.component.ts

```diff
@@ -404,180 +98,179 @@ export class SprkMastheadComponent implements AfterContentInit {
    */
   @Input()
   idString: string;
-  /**
-   * Expects an array of
-   * [ISprkBigNavLink](https://github.com/sparkdesignsystem/spark-design-system/blob/main/angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.interfaces.ts)
-   *  to be
-   * used to create the Big Navigation of
-   * the Masthead component.
-   */
-  @Input()
-  bigNavLinks: ISprkBigNavLink[];
-  /**
-   * Expects a [ISprkNarrowSelector](https://github.com/sparkdesignsystem/spark-design-system/blob/main/angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.interfaces.ts)
-   *  object that
-   * represents dropdown choices inside the dropdown
-   * rendered in the Narrow Navigation.
-   */
-  @Input()
-  narrowSelector: ISprkNarrowSelector;
-  /**
-   * A string that is used to set the `id` on the narrow nav
-   * and the `aria-controls` for the menu trigger button.
-   */
-  @Input()
-  narrowNavId = _.uniqueId(`sprk_masthead_narrow_nav_`);
+
+  @ContentChild(SprkMastheadNavCollapsibleDirective, {
+    static: false,
+  })
+  collapsibleNavDirective: SprkMastheadNavCollapsibleDirective;
+
+  @ViewChild('mastheadMenuContainer', { static: false })
+  mastheadMenuContainer: ElementRef;
+
+  @ContentChild(SprkMastheadBrandingDirective, {
+    static: false,
+    read: ElementRef,
+  })
+  branding: ElementRef;
 
   /**
-   * @ignore
-   */
-  iconType = 'chevron-down';
-  /**
-   * @ignore
+   * Keeps state of if the page has been scrolled.
    */
-  componentID = _.uniqueId();
+  isPageScrolled = false;
   /**
-   * @ignore
+   * Keeps state of if the Masthead is on a narrow viewport.
    */
-  controls_id = `sprk-narrow-navigation-item__${this.componentID}`;
+  isNarrowViewport = false;
   /**
-   * @ignore
+   * Keeps state of the scroll direction in order to apply
+   * CSS classes or not.
    */
-  isScrolled = false;
+  currentScrollDirection = 'up';
   /**
-   * @ignore
+   * Keeps state of if the Masthead is hidden in order to apply
+   * CSS classes or not.
    */
-  isNarrowLayout = false;
+  isMastheadHidden = false;
   /**
-   * @ignore
+   * Keeps state of the viewport on resize.
    */
-  scrollDirection = 'up';
+  isNarrowViewportOnResize = false;
   /**
-   * @ignore
+   * Keeps state of the scroll position in
+   * order to apply CSS classes or not.
    */
-  isHidden = false;
+  currentScrollPosition = 0;
+
   /**
-   * @ignore
+   * Represents the initial state of the
+   * collapsible nav of the Masthead component.
    */
-  isNarrowOnResize = false;
+  isCollapsibleNavOpen = false;
   /**
-   * @ignore
+   * Throttles the updateScrollDirection function to prevent
```

Comment:
I removed these `@ignores` and added comments so these are documented in our storybook. This allows others to read what the masthead functionality is.